NSS-Day-Cohort-42/rare-server-media-moguls
# Cb post join category

#### Changes Made
1. changed the get all posts method in post request so that user and category information is joined with post
2. created posts by category id method in post request. am not using it but kept it in because it will useful later
3. added necessary changes in request_handler.py
​
#### Steps to Review
1. Checkout this branch locally and run the application.
```
git fetch --all
git checkout cb-post-join-category
```
2. Run the application.
```
npm start
```
3. Open a new Terminal tab (⌘T) and navigate to the server directory.
```
..
server
```
4. Initialize virtual environment, and run the server.
```
pipenv shell
watchgod request_handler.main
```
5. Test app functionality. 
> create the proper tables in sql for Posts, Users, Categories
> seed with necessary data: https://gist.github.com/heymonicakay/9176ae9b672608777813b237cda403be
> visit Postman and use http://localhost:8000/post, should return all posts with user display name and category 
> use in conjunction with cb-post-list branch to make sure the proper data shows up
6. View code file.
> Confirm file modifications are present as indicated above.
> Confirm no unused code or extraneous comments exist.
